va_explorer/users/views.py

```diff
@@ -16,9 +18,23 @@
 from ..utils.mixins import CustomAuthMixin, UserDetailViewMixin
 from .forms import ExtendedUserCreationForm, UserChangePasswordForm, UserSetPasswordForm, UserUpdateForm
 from ..va_data_management.models import VaUsername
+from ..va_logs.logging_utils import write_va_log
+import logging
 
 User = get_user_model()
-
+LOGGER = logging.getLogger("event_logger")
+
+# track login
+@receiver(user_logged_in)
+def login_logger(request, user, **kwargs):
+    username = user.__dict__.get('username', user.email)
```

Comment:
I think you can do `getattr(user, 'username', user.email)` instead of accessing the __dict__ 

Although isn't username a required field, so it should always exist for the user?

Reply:
Good call. I wasn't sure if it was required at the time so did a soft access just to be safe. But after review, seems like by default we assign user's email to username when none is provided so should be safe to do a hard access.